### PR TITLE
test: move licence check to static_analysis test suite

### PR DESCRIPTION
**Pull request by wallyworld**
The separate check for licence violations is moved to the bash tests suite. This allows the separate pre merge check to be deleted.

Also add a new action that produces a json file representing
what checks should be run. This will be used later.

## QA steps

Look at the pre merge checks.

## Links

**Jira card:** [JUJU-6356](https://warthogs.atlassian.net/browse/JUJU-6356)

[JUJU-6356]: https://warthogs.atlassian.net/browse/JUJU-6356?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/migrate.yml [hunk 1]
```diff
@@ -36,7 +36,7 @@ jobs:
       - name: Set flag
         id: set-flag
         run: |
-          echo "SHOULD_RUN=steps.should-run.outputs.should_run" >> $GITHUB_OUTPUT
+          echo "SHOULD_RUN=steps.should-run.outputs.should-run" >> $GITHUB_OUTPUT
 
   migrate:
     needs: [should-run]
```
Comment (hpidcock) [inline .github/workflows/migrate.yml:42]: I don't think we can make the individual matrix jobs required if they only run sometimes. Might be easier to make a post-migrate job required.

```
post-migrate:
    needs: [migrate]
    if: always() && !failure() && !cancelled()
    runs-on: ubuntu-latest
    steps:
    - name: Success
        run: echo success
```

Comment (hpidcock) [reply .github/workflows/migrate.yml:42]: Never tried this, but I think it might work.

**Review by hpidcock — APPROVED** [review]
Looks ok.

File: .github/workflows/migrate.yml [hunk 2]
```diff
@@ -53,7 +53,7 @@ jobs:
         client: ['source', 'target']
 
     steps:
-      - name: Checkout code
+      - name: Checkout codeMakef
```
[the remark this answers is not in the document]
Comment (wallyworld) [reply .github/workflows/migrate.yml:56]: lol, wrong window got the keyboard input

File: .github/workflows/migrate.yml (hunk 1 above)
Comment (wallyworld) [reply .github/workflows/migrate.yml:42]: Yeah, the matrix jobs screw it up. I noticed it last week with the initial PR on this.

**Pull request merged**